frontend/app/components/redesign/components/ToolsWalletAddress.tsx

```diff
@@ -8,13 +8,29 @@ import { SVGRefresh, SVGSpinner } from '~/assets/svg'
 import { toolState, toolActions } from '~/stores/toolStore'
 import type { ElementErrors } from '~/lib/types'
 import { Heading5 } from '../Typography'
+import { useUI } from '~/stores/uiStore'
 
 export const ToolsWalletAddress = () => {
   const snap = useSnapshot(toolState)
+  const { actions: uiActions } = useUI()
   const [error, setError] = useState<ElementErrors>()
   const [isLoading, setIsLoading] = useState(false)
   const generatedId = useId()
   const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    const walletInputApi = {
+      focus: () => {
+        if (inputRef.current) {
+          inputRef.current.focus()
+          const length = inputRef.current.value.length
+          inputRef.current.setSelectionRange(length, length)
+        }
+      }
+    }
+    uiActions.registerWalletInput(walletInputApi)
```

Comment:
So it'll get unregistered by `useEffect`?
```suggestion
    return uiActions.registerWalletInput(walletInputApi)
```